File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/about/AboutTitleBuilder.java
package com.lasthopesoftware.bluewater.about;

import android.content.Context;
import com.lasthopesoftware.bluewater.R;
import com.namehillsoftware.lazyj.AbstractSynchronousLazy;
import com.namehillsoftware.lazyj.CreateAndHold;

public class AboutTitleBuilder implements BuildAboutTitle {
	private final CreateAndHold<String> lazyAboutTitle;

	public AboutTitleBuilder(Context context) {
		lazyAboutTitle = new AbstractSynchronousLazy<String>() {
			@Override
			protected String create() {
				return String.format(
					context.getString(R.string.title_activity_about),
					context.getString(R.string.app_name));
			}
		};
	}

	@Override
	public String buildTitle() {
		return lazyAboutTitle.getObject();
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/about/AboutTitleBuilder.kt
package com.lasthopesoftware.bluewater.about

import android.content.Context
import com.lasthopesoftware.bluewater.R

class AboutTitleBuilder(context: Context) : BuildAboutTitle {
	private val lazyAboutTitle by lazy {
		context
			.getString(R.string.title_activity_about)
			.format(context.getString(R.string.app_name))
	}

	override fun buildTitle(): String = lazyAboutTitle
}
